components/lib/datatable/ColumnFilter.js

```diff
@@ -563,10 +563,6 @@ export const ColumnFilter = React.memo((props) => {
     };
 
     const createClearButton = () => {
-        if (!showMenuButton()) {
```

Comment:
This should check !showClearButton which doesn't exist and needs to be created next to the showMenuButton.

Reply:
> This should check !showClearButton which doesn't exist and needs to be created next to the showMenuButton.

Doesn't [this line](https://github.com/primefaces/primereact/pull/5803/files/85a85ca43f303a67a5f91781c7aae86624602ec7#diff-b816aae7f5071fd81bcae0d885dbe0b2bbae790ce43618e2a26e2b52e6f076adR575) (line 579/575) already check if the clear button should be rendered or not?

Reply:
It does but its now not consistent with the MenuButton logic.  It should check it first and short circuit the whole method. Its cleaner and consistent. I am big on consistency!

Reply:
I can fix it if you don't have time.  Just let me know.

Reply:
@melloware I know @cassenhp is sick and I do not have access to their fork - so that would be appreciated. Otherwise I can create my own fork and make the changes needed.

Reply:
@lbathp i will take care of it!  Thanks!